stacker/config/constructors.py

```diff
@@ -0,0 +1,39 @@
+import json
+import subprocess
+import yaml
+
+
+def get_vaulted_value(value):
+    """Get the specified value from vault.
+
+    Vaulted field types should use the following format:
+
+        <path within vault>@<key within stored vault data>
+
+    For example:
+
+        # We've set the secret/hello key with the following
+        $ vault write secret/hello value=world
+
+        # In stacker we would reference "world" with the following
+        conf_key: !vault secret/hello@value
+
+        # The above would resolve to
+        conf_key: world
+
+    """
+    if '@' not in value:
```

Comment:
I'd just catch the valueError that will come if you try to split a field and it doesn't have an @ in it.